### PR TITLE
[ui] Polish runs feed UI

### PR DESCRIPTION
**Pull request by bengotow**
## Summary & Motivation

Related: https://linear.app/dagster-labs/issue/FE-515/backfills-and-runs-consolidation-ui-frontend

This is a collection of small tweaks to improve the UI/UX of the runs feed pulled from this doc of dogfooding feedback: https://www.notion.so/dagster/new-Runs-page-1-9-dogfooding-12918b92e462807688fee0e644a34641

- UI does not reset to page 1 when you change filters
- “Target” column occupies more space than necessary
- When paginating, should reset scroll to top
- Might be good to use middle-truncate or end-truncate on single assets/jobs in the Target column
- Column cuts off “Launched by” content
- “Automation condition” tag tooltip is applied to the entire width of the cell instead of just the tag, probably need a wrapper div on the
- During Loading, the header of the table is shown, then it disappears if there are no results
- 26 runs per page?
- Run page header vertical padding is lost on narrow viewport
- Run timeline: use monospace font for run ID

## How I Tested These Changes

I tested all of these manually using the runs feed page and other impacted pages (for the middle truncate items)

### CODE REVIEW

File: js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx [hunk 1]
```diff
@@ -25,8 +25,9 @@ export const PageHeader = (props: Props) => {
     >
       {title && (
         <Box
+          padding={{vertical: 8}}
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx:28]: This is a global change but I verified it on a bunch of pages - the padding only takes effect when the height of the content in this box exceeds 36px, which is only the case when it is wrapping to two lines.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx [hunk 2]
```diff
@@ -317,17 +317,19 @@ export const AssetNodeOverview = ({
       </AttributeAndValue>
 
       <AttributeAndValue label="Code location">
-        <Box flex={{direction: 'column'}}>
+        <Box flex={{direction: 'column', alignItems: 'flex-start', gap: 8}}>
```
Comment (bengotow) [inline js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx:320]: This is a drive-by fix for this scenario:

image


after:

image

File: js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx (hunk 1 above)
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx:28]: Is it worth making this its own PR? It might be annoying to have to do a revert if this breaks something unexpectedly.

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx (hunk 2 above)
Comment (hellendag) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx:320]: I feel like it would be nice to have a mini-alert component, these always feel too big for sidebars.

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunsFeedRow.tsx [hunk 3]
```diff
@@ -104,6 +104,12 @@ export const RunsFeedRow = ({
             flex={{direction: 'row', alignItems: 'center', wrap: 'wrap'}}
             style={{gap: '4px 8px', lineHeight: 0}}
           >
+            {entry.__typename === 'PartitionBackfill' ? (
+              <Tag intent="none">
+                <span>Backfill</span>
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/runs/RunsFeedRow.tsx:109]: `span` wrapper necessary?

File: js_modules/dagster-ui/packages/ui-core/src/runs/useRunsFeedEntries.tsx [hunk 4]
```diff
@@ -11,9 +11,9 @@ import {gql, useQuery} from '../apollo-client';
 import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
 import {RunsFilter} from '../graphql/types';
 
-const PAGE_SIZE = 25;
+const PAGE_SIZE = 29;
```
Comment (hellendag) [inline js_modules/dagster-ui/packages/ui-core/src/runs/useRunsFeedEntries.tsx:14]: Why 29?

**Review by hellendag — APPROVED** [review]

File: js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-components/src/components/PageHeader.tsx:28]: Ahh I think you're right, I'll break this out into a separate PR

File: js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx (hunk 2 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/assets/AssetNodeOverview.tsx:320]: Yeah I agree, there's at least one that appears in the asset graph's sidebar and I think in that scenario we place it at the top with no whitespace around it all. Will see if Josh has thoughts about this in a follow-up

File: js_modules/dagster-ui/packages/ui-core/src/runs/RunsFeedRow.tsx (hunk 3 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/runs/RunsFeedRow.tsx:109]: Looks like it's not! Will remove

File: js_modules/dagster-ui/packages/ui-core/src/runs/useRunsFeedEntries.tsx (hunk 4 above)
Comment (bengotow) [reply js_modules/dagster-ui/packages/ui-core/src/runs/useRunsFeedEntries.tsx:14]: Ahh this is because of a bug on the graphql side that causes it to return 1 more row than requested. Let me see if i can find the root cause and fix it there so this is less janky
